hyperspy/utils/peakfinders2D.py

```diff
@@ -0,0 +1,455 @@
+
+    def local_std(image, radius):
+        """Calculates rolling standard deviation over a circular kernel."""
+        return _local_stat(image, radius, np.std)
+
+    def single_pixel_desensitize(image):
+        """Reduces single-pixel anomalies by nearest-neighbor smoothing."""
+        kernel = np.array([[0.5, 1, 0.5], [1, 1, 1], [0.5, 1, 0.5]])
+        smoothed_image = generic_filter(image, np.mean, footprint=kernel)
+        return smoothed_image
+
+    def stat_binarise(image):
+        """Peaks more than one standard deviation from the mean set to one."""
+        image_rolling_mean = local_mean(image, window_radius)
+        image_rolling_std = local_std(image, window_radius)
+        image = single_pixel_desensitize(image)
+        binarised_image = np.zeros(image.shape)
+        stat_mask = image > (image_rolling_mean + alpha * image_rolling_std)
+        binarised_image[stat_mask] = 1
+        return binarised_image
+
+    def smooth(image):
+        """Image convolved twice using a uniform 3x3 kernel."""
+        image = uniform_filter(image, size=3)
+        image = uniform_filter(image, size=3)
+        return image
+
+    def half_binarise(image):
+        """Image binarised about values of one-half intensity."""
+        binarised_image = np.where(image > 0.5, 1, 0)
+        return binarised_image
+
+    def separate_peaks(binarised_image):
+        """Identify adjacent 'on' coordinates via DBSCAN."""
+        bi = binarised_image.astype('bool')
+        coordinates = np.indices(bi.shape).reshape(2, -1).T[
+            bi.flatten()]
+        db = DBSCAN(2, 3)
+        peaks = []
+        labeled_points = db.fit_predict(coordinates)
+        for peak_label in list(set(labeled_points)):
+            peaks.append(coordinates[labeled_points == peak_label])
+        return peaks
+
+    def _peak_find_once(image):
+        """Smooth, binarise, and find peaks according to main algorithm."""
+        image = smooth(image)
+        image = half_binarise(image)
+        peaks = separate_peaks(image)
+        return image, peaks
+
+    def stat_peak_finder(image):
+        """Find peaks in image. Algorithm stages in comments."""
+        image = normalize(image)  # 1
+        image = stat_binarise(image)  # 2, 3
+        n_peaks = np.infty  # Initial number of peaks
+        image, peaks = _peak_find_once(image)  # 4-6
+        m_peaks = len(peaks)  # Actual number of peaks
+        while (n_peaks - m_peaks) / n_peaks > convergence_ratio:  # 8
+            n_peaks = m_peaks
+            image, peaks = _peak_find_once(image)
+            m_peaks = len(peaks)
+        peak_centers = np.array(
+            [np.mean(peak, axis=0) for peak in peaks])  # 7
+        return peak_centers
+
+    return clean_peaks(np.round(stat_peak_finder(z)).astype(int))
+
+
+def find_peaks_dog(z, min_sigma=1., max_sigma=50., sigma_ratio=1.6,
+                   threshold=0.2, overlap=0.5, exclude_border=False):
```

Comment:
Why adding `exclude_border` if this is not used?

Reply:
Thanks - it was supposed to be used as a kwarg for the skimage blob_dog method - I've made it actually use this and updated the skimage dependency to >=0.15 as that is when this kwarg was introduced.